File: Components/Strata.Client.Core/main/typescript/Service/StandardResponseProcessor.ts
import {IResponseProcessor} from "./IResponseProcessor";
import {IResponse} from "./IResponse";
import {ServiceError} from "./ServiceError";

export
class StandardResponseProcessor
    implements IResponseProcessor
{
    process<R>(response: IResponse): Promise<R>
    {
        switch (response.getStatus())
        {
            case 200: // OK
            case 500: //INTERNAL_SERVER_ERROR
                return response.readEntity<R>();

            default:
                throw new ServiceError(response);
        }
    }
}

File: Components/Strata.Client.Core/main/java/strata/client/core/service/StandardResponseProcessor.java
//////////////////////////////////////////////////////////////////////////////
// StandardResponseProcessor.java
//////////////////////////////////////////////////////////////////////////////

package strata.client.core.service;

public
class StandardResponseProcessor
    implements IResponseProcessor
{

    @Override
    public <R> R
    process(Class<R> replyType,IResponse response)
    {
        try
        {
            switch (response.getStatusAsEnum())
            {
                case OK:
                case INTERNAL_SERVER_ERROR:
                    return response.readEntity(replyType);

                default:
                    throw new ServiceException(response);
            }
        }
        finally
        {
            response.close();
        }
    }


}

//////////////////////////////////////////////////////////////////////////////
